janus_core/cli/phonons.py

```diff
@@ -184,18 +184,17 @@ def phonons(
         raise ValueError("'fmax' must be passed through the --fmax option")
     minimize_kwargs["fmax"] = fmax
 
-    # Set default supercell
-    if not supercell:
-        supercell = [2, 2, 2]
+    # Convert supercell string to list
+    supercell = supercell.split("x")
 
-    # Convert single value to list for supercell matrix
-    if len(supercell) == 1:
-        supercell *= 3
+    # Validate supercell list
     if len(supercell) != 3:
-        raise ValueError(
-            """Please pass the lattice vectors via either --supercell x --supercell y \
---supercell z, or --supercell x."""
-        )
+        raise ValueError("Please pass three lattice vectors in the form 1x2x3")
+
+    if not all(vector.isdigit() for vector in supercell):
+        raise ValueError("Please pass lattice vectors as integers in the form 1x2x3")
+
+    supercell = [int(vector) for vector in supercell]
```

Comment:
Minor stylistic thing
```suggestion
    try:
        supercell = [int(x) for x in supercell.split("x")]
    except ValueError as exc:
        raise ValueError("Please pass lattice vectors as integers in the form 1x2x3") from exc

    # Validate supercell list
    if len(supercell) != 3:
        raise ValueError("Please pass three lattice vectors in the form 1x2x3")
```

Reply:
Ah I originally considered int(x), but didn't realise something like `int("2.4")` throws an error, since `int(2.4)` doesn't. That's good to know